x/pairing/keeper/msg_server_relay_payment.go

```diff
@@ -392,3 +391,102 @@ func appendRelayPaymentDetailsToEvent(from map[string]string, uniqueIdentifier u
 	}
 	return to
 }
+
+// distributeRewards is the main function for reward distribution for providers and delegators
+func (k Keeper) distributeRewards(ctx sdk.Context, providerAddr sdk.AccAddress, chainID string, block uint64, totalReward math.Int) error {
+	epoch, _, err := k.epochStorageKeeper.GetEpochStartForBlock(ctx, block)
+	if err != nil {
+		return utils.LavaFormatError("could not calculate provider reward by delegations", err,
```

Comment:
the first string repeats throughout this function. use a const?

Reply:
done. see https://github.com/lavanet/lava/commit/c23f6d56950de9bd0d143d7fb62471402a5354e2